daml-foundations/daml-ghc/damldoc/src/DA/Daml/GHC/Damldoc/Types.hs

```diff
@@ -30,7 +30,7 @@ newtype Modulename = Modulename { unModulename :: Text }
     deriving newtype (Eq, Ord, Show, ToJSON, FromJSON, IsString)
 
 -- | Type expression, possibly a (nested) type application
-data Type = TypeApp Typename [Type] -- ^ Type application
+data Type = TypeApp !(Maybe Anchor) !Typename [Type] -- ^ Type application
```

Comment:
Including the anchor here seems slightly weird but I’m not really sure what a better place would be either.

Reply:
I was going to add a Modulename at first, but then realised I'd need to disambiguate, e.g., templates from classes and choices and other types. So I came to the conclusion that unless I was willing to replicate all that logic, or have the Ghc API bleed into the rest of damldocs, the thing I really want is to know the anchor so I can link the type to the right place when it's time to render it.